src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNameParser.cs

```diff
@@ -170,5 +170,27 @@ internal static bool TryParseAnonymousTypeParameterName(string typeParameterName
             propertyName = null;
             return false;
         }
+
+        internal static string GetOriginalLocalVariableName(string name)
```

Comment:
>GetOriginalLocalVariableName [](http://example.com/codeflow?start=31&length=28)

Consider moving these two methods to ResultProvider if `Formatter` is the only caller that requires these. (We already have a couple of uses of `GeneratedNameParser.TryParseGeneratedName()` in the EE that are specific to scenarios there.)

Reply:
See https://github.com/dotnet/roslyn/pull/60522#discussion_r842181021
Should I leave it this in Formatter or in `GeneratedNameParser` @tmat

Reply:
Moved to CSharpFormatter